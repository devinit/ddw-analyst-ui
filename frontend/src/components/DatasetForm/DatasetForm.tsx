import { Formik, FormikProps } from 'formik';
import * as React from 'react';
import { Button, ButtonToolbar, Form } from 'react-bootstrap';
import * as Yup from 'yup';
import { Dataset } from '../../types/datasets';
import { MaterialFormGroup } from '../MaterialFormGroup';

const schema = Yup.object().shape({
  title: Yup.string().required('Title is required!'),
  description: Yup.string().required('Description is required!'),
});
const initialValues: Partial<Dataset> = { title: '', description: '' };

export const DatasetForm: React.SFC = () => {
  const [focused, setFocused] = React.useState('');
  const onSuccess = () => {
    console.log('Successful'); //tslint:disable-line
  };
  const onFocus = (event: React.FocusEvent<HTMLInputElement>) => {
    setFocused(event.currentTarget.name);
  };
  const onBlur = (event: React.FocusEvent<HTMLInputElement>) => {
    if (event.currentTarget.name === focused) {
      setFocused('');
    }
  };

  return (
    <Formik validationSchema={schema} initialValues={initialValues} onSubmit={onSuccess}>
      {({ errors, handleChange, handleSubmit, touched, values }: FormikProps<Dataset>) => (
        <Form className="form" noValidate onSubmit={handleSubmit} data-testid="dataset-form">
          <MaterialFormGroup
            label="Title"
            md="8"
            id="title"
            name="title"
            required
            value={values.title}
            onChange={handleChange}
            onFocus={onFocus}
            onBlur={onBlur}
            touched={touched.title}
            errors={errors.title}
            focused={focused === 'title'}
          />

          <MaterialFormGroup
            label="Description"
            as="textarea"
            id="description"
            name="description"
            required
            value={values.description}
            onChange={handleChange}
            onFocus={onFocus}
            onBlur={onBlur}
            touched={touched.description}
            errors={errors.description}
            focused={focused === 'description'}
          />

          <MaterialFormGroup
            label="Publication"
            as="textarea"
            id="publication"
            name="publication"
            required
            value={values.publication}
            onChange={handleChange}
            onFocus={onFocus}
            onBlur={onBlur}
            touched={touched.publication}
            errors={errors.publication}
            focused={focused === 'publication'}
          />
          <ButtonToolbar>
            <Button className="btn-rose" type="submit">
              Save
            </Button>
          </ButtonToolbar>
        </Form>
      )}
    </Formik>
  );
};

export { DatasetForm as default };
