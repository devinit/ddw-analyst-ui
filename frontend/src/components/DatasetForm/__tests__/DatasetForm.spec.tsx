/**
 * @jest-environment jsdom
 */
import '@testing-library/jest-dom/extend-expect';
// import { render } from '@testing-library/react';
import * as React from 'react';
import * as TestRenderer from 'react-test-renderer';
import { DatasetForm } from '../DatasetForm';

test('renders the form correctly', () => {
  const renderer = TestRenderer.create(<DatasetForm/>).toJSON();

  expect(renderer).toMatchSnapshot();
});
