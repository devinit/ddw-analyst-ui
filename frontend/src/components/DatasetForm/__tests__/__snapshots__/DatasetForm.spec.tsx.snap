// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the form correctly 1`] = `
<form
  className="form"
  data-testid="dataset-form"
  noValidate={true}
  onSubmit={[Function]}
>
  <div
    className="row"
  >
    <div
      className="col-md-8"
    >
      <div
        className="bmd-form-group form-group"
        data-testid="material-form-group"
      >
        <label
          className="bmd-label-floating form-label"
          htmlFor="title"
        >
          Title
        </label>
        <input
          className="form-control"
          id="title"
          name="title"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={true}
        />
        <div
          className="invalid-feedback"
          data-testid="material-form-control-feedback"
        />
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <div
        className="bmd-form-group form-group"
        data-testid="material-form-group"
      >
        <label
          className="bmd-label-floating form-label"
          htmlFor="description"
        >
          Description
        </label>
        <textarea
          className="form-control"
          id="description"
          name="description"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={true}
        />
        <div
          className="invalid-feedback"
          data-testid="material-form-control-feedback"
        />
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <div
        className="bmd-form-group form-group"
        data-testid="material-form-group"
      >
        <label
          className="bmd-label-floating form-label"
          htmlFor="publication"
        >
          Publication
        </label>
        <textarea
          className="form-control"
          id="publication"
          name="publication"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={true}
        />
        <div
          className="invalid-feedback"
          data-testid="material-form-control-feedback"
        />
      </div>
    </div>
  </div>
  <div
    className="btn-toolbar"
    role="toolbar"
  >
    <button
      className="btn-rose btn btn-primary"
      disabled={false}
      type="submit"
    >
      Save
    </button>
  </div>
</form>
`;
